src/templates/datagrid.latte

```diff
@@ -26,25 +26,29 @@
 			 *}
 			{if $control->hasOuterFilterRendering()}
 				<div class="row row-filters">
-					{var $i = 0}
-					<div class="col-sm-6" n:foreach="$filters as $f">
+					{foreach $filters as $f}
 						{**
-						 * Each fitler is rendered separately in its own template
+						 * Do not display the filter if the column is hidden
 						 *}
-						{var $filter_block = 'filter-' . $f->getKey()}
-						{var $filter_type_block = 'filtertype-' . $f->getType()}
-
-						{ifset #$filter_block}
-							{include #$filter_block, filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
-						{else}
-							{ifset #$filter_type_block}
-								{include #$filter_type_block, filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
-								{else}
-									{include $f->getTemplate(), filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
+						{continueIf !array_key_exists($f->getKey(), $columns)}
```

Comment:
Hello, could you use `!isset($columns[$f->getKey()]`? It is faster and should not cause any issues in this case. Also you could possibly get filter key from the foreach above.